src/com/SocketCommunication.cpp

```diff
@@ -199,21 +183,14 @@ void SocketCommunication::requestConnection(std::string const &acceptorName,
   TRACE(acceptorName, requesterName);
   assertion(not isConnected());
 
-  std::string address;
-  const std::string addressFileName("." + requesterName + "-" + acceptorName + ".address");
+  ConnectionInfoReader conInfo(acceptorName, requesterName, _addressDirectory);
+  std::string address = conInfo.read();
+  DEBUG("Request connection to " << address);
+  std::string ipAddress  = address.substr(0, address.find(":"));
+  std::string portNumber = address.substr(ipAddress.length() + 1, address.length() - ipAddress.length() - 1);
+  _portNumber = static_cast<unsigned short>(std::stoi(portNumber));
```

Comment:
As the port is unsigned, so lets convert the string to an unsigned straight away.
```suggestion
  _portNumber = static_cast<unsigned short>(std::stoul(portNumber));
```

This could potentially fail in case the file contains rubbish.
Maybe this entire parsing belongs into a `try` block?

Reply:
I have adopted the change to `stoul`. Regarding the try block: If the file contains rubbish, there is no way we can continue in any sensible way, so we need to fail sooner or later, as the error can not be handled gracefully. Furthermore, the files are written be precice and read by precice, if there is rubbish in there, there is a problem with the underlying system, which can produce **any** kind of error, ranging from segfaults to formatting of your hard disk.